tool/src/js/gen.rs

```diff
@@ -66,15 +73,37 @@ impl<'ctx, 'tcx> TyGenContext<'ctx, 'tcx> {
     /// A wrapper for `borrow_mut`ably inserting new imports.
     ///
     /// I do this to avoid borrow checking madness.
-    pub(super) fn add_import(&self, import_str: String) {
-        self.imports.borrow_mut().insert(import_str);
+    pub(super) fn add_import(
+        &self,
+        import_str: Cow<'tcx, str>,
+        import_file: Option<Cow<'tcx, str>>,
+        usage: ImportUsage,
+    ) {
+        self.imports.borrow_mut().insert(ImportInfo {
```

Comment:
thought: in the Both case should we insert twice instead? Otherwise this will lead to duplicate imports if you push a Both and a Typescript, yes?

Reply:
Yeah, I think this enum should act as a flag instead of being distinct for each.

Reply:
As for duplicate imports, I don't think this will happen?

I have custom definitions for Ord, PartialOrd, and PartialEq (gen.rs line 691) that only check the name of the type being imported, not the file or usage.

A problem that arises is that I'm not sure if the usage will get updated on adding a new import. For instance, from "custom_import", ImportUsage::Typescript to "custom_import", ImportUsage::Both.

I'll look into it.

Reply:
Yeah, a custom Ord impl has the other problem you mentioned. You should do one of:

 - Multiple maps
 - Multiple inserts into the same map
 - Make the import usage part of the value, not the key, and have the insertion process update it.


IMO multiple maps is easiest.

Reply:
I think you're right, I'll create two maps for each import.